packages/password/src/index.mts

```diff
@@ -1,39 +1,110 @@
-import type { Prompt } from '@inquirer/type';
-import input from '@inquirer/input';
+import {
+  createPrompt,
+  useState,
+  useKeypress,
+  usePrefix,
+  isEnterKey,
+  type PromptConfig,
+} from '@inquirer/core';
 import chalk from 'chalk';
 import ansiEscapes from 'ansi-escapes';
+import { Context } from '@inquirer/type';
 
-type InputConfig = Parameters<typeof input>[0];
-type PasswordConfig = Omit<InputConfig, 'transformer' | 'default'> & {
+type PasswordConfig = PromptConfig<{
   mask?: boolean | string;
+  validate?: (value: string) => boolean | string | Promise<string | boolean>;
+  allowShowPassword?: boolean;
+}>;
+
+const transformer = (
+  str: string,
+  mask: string | boolean | undefined,
+  { isFinal }: { isFinal: boolean },
+) => {
+  if (mask) {
+    const maskChar = typeof mask === 'string' ? mask : '*';
+    return maskChar.repeat(str.length);
+  }
+
+  if (!isFinal) {
+    return `${chalk.dim('[input is masked]')}${ansiEscapes.cursorHide}`;
+  }
+
+  return '';
 };
 
-const password: Prompt<string, PasswordConfig> = (config, context) => {
+export default (config: PasswordConfig, context?: Context | undefined) => {
   if ('transformer' in config) {
     throw new Error(
       'Inquirer password prompt do not support custom transformer function. Use the input prompt instead.',
     );
   }
 
-  return input(
-    {
-      ...config, // Make sure we do not display the default password
-      default: undefined,
-      transformer(str: string, { isFinal }: { isFinal: boolean }) {
-        if (config.mask) {
-          const maskChar = typeof config.mask === 'string' ? config.mask : '*';
-          return maskChar.repeat(str.length);
-        }
+  return createPrompt<string, PasswordConfig>((_config, done) => {
+    const { validate = () => true, allowShowPassword } = _config;
+    const [status, setStatus] = useState<string>('pending');
+    const [errorMsg, setError] = useState<string | undefined>(undefined);
+    const [value, setValue] = useState<string>('');
+    const [togglePassword, setTogglePassword] = useState<boolean>(false);
```

Comment:
I think `isMasked` would be a better name than toggle. "toggle" being an action verb makes it feel like it is a function.

Reply:
I agree. I renamed `togglePassowrd` to `isMasked` and `setTogglePassword` to `setIsMasked`. For consistency, `isMasked` is set to `true` by default, since the password starts masked, and is set to false when password is shown